src/Sql/Sql/VirtualCluster/Cmdlet/GetAzureSqlVirtualCluster.cs

```diff
@@ -0,0 +1,132 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.VirtualCluster.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Sql.VirtualCluster.Cmdlet
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlVirtualCluster", DefaultParameterSetName = GetByResourceGroupParameterSet)]
+    [OutputType(typeof(AzureSqlVirtualClusterModel))]
+    public class GetAzureSqlVirtualCluster : VirtualClusterCmdletBase
+    {
+        protected const string GetByNameAndResourceGroupParameterSet =
+            "GetVirtualClusterByNameAndResourceGroup";
+
+        protected const string GetByResourceGroupParameterSet =
+            "GetVirtualClusterByResourceGroup";
+
+        protected const string GetByResourceIdParameterSet =
+            "GetVirtualClusterByResourceId";
+
+        /// <summary>
+        /// Gets or sets the name of the virtual cluster.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The name of the virtual cluster.")]
+        [Alias("VirtualClusterName")]
+        [ResourceNameCompleter("Microsoft.Sql/virtualClusters", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the name of the resource group.
+        /// </summary>
+        [Parameter(ParameterSetName = GetByNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The name of the resource group.")]
+        [Parameter(ParameterSetName = GetByResourceGroupParameterSet,
+            Mandatory = false,
+            Position = 1,
```

Comment:
@milostod please remove the `Position` property from optional parameters

Reply:
Done.